Dockerfile

```diff
@@ -27,90 +29,127 @@
 FROM ubuntu:22.04 AS os
 
 # tzinfo hangs without this
-ARG DEBIAN_FRONTEND=noninteractive
+ARG DEBIAN_FRONTEND='noninteractive'
 
 # ---------------------------------------------------------------------
 # build
 # ---------------------------------------------------------------------
 
 FROM os AS build
 
+# limit cpus so don't run out of memory on local machine
+# symptom: get error - "c++: fatal error: Killed signal terminated program cc1plus"
+# can turn off if building in cloud
+ARG CONAN_CPU_COUNT=10
+
+ARG WITH_RUBY='True'
+
+# set some variables
+ENV PATH="$HOME/venv3.9/bin:$PATH"
+ENV CONAN_PROFILE='conan/profiles/docker'
+
 # update os and add dependencies
 # note: Dockerfiles run as root by default, so don't need sudo
-RUN apt-get clean \
-  && apt-get update \
+RUN apt-get update \
   && apt-get install -y \
-  build-essential python3.9 python3-pip git cmake make ruby rake autoconf automake \
-  && pip install conan -v "conan==1.59.0"
+    autoconf \
+    automake \
+    build-essential \
+    cmake \
+    git \
+    python3 \
+    python3-pip \
+    rake \
+    ruby \
+  && rm -rf /var/lib/apt/lists/* \
+  && pip install conan -v 'conan==2.0.9'
 
 # make an agent directory and cd into it
 WORKDIR /root/agent
 
 # bring in the repo contents, minus .dockerignore files
 COPY . .
 
-# set some variables
-ENV PATH=$HOME/venv3.9/bin:$PATH
-ENV CONAN_PROFILE=conan/profiles/docker
-ENV WITH_RUBY=True
-ENV WITH_TESTS=False
-
-# limit cpus so don't run out of memory on local machine
-# symptom: get error - "c++: fatal error: Killed signal terminated program cc1plus"
-# can turn off if building in cloud
-ENV CONAN_CPU_COUNT=1
-
 # make installer
-RUN conan export conan/mqtt_cpp \
-  && conan export conan/mruby \
-  && conan install . -if build --build=missing \
-  -pr $CONAN_PROFILE \
-  -o build_tests=$WITH_TESTS \
-  -o run_tests=$WITH_TESTS \
-  -o with_ruby=$WITH_RUBY
-
-# compile source (~20mins - 4hrs for qemu)
- RUN conan build . -bf build
+RUN conan profile detect \
+  && conan install . \
+    --build=missing \
+    -c "tools.build:jobs=$CONAN_CPU_COUNT" \
+    -c tools.build:skip_test=True \
+    -o "with_ruby=$WITH_RUBY" \
+    -pr "$CONAN_PROFILE" \
+  && conan create . \
+    --build=missing \
+    -c "tools.build:jobs=$CONAN_CPU_COUNT" \
+    -c tools.build:skip_test=True \
+    -o agent_prefix=mtc \
+    -o cpack=True \
+    -o "with_ruby=$WITH_RUBY" \
+    -o zip_destination=/root/agent \
+    -pr "$CONAN_PROFILE" \
+    -tf ''
```

Comment:
I presume you have verified this is what you want to do in `Dockerfile` to build the agent.